doc: new single-click setup onboarding (#29)
elastic/universal-profiling-documentation: README.md

Before setting up Universal Profiling, make sure you meet the following requirements:

- An Elastic stack deployment on [Elastic Cloud](http://cloud.elastic.co) at version 8.7.0 or higher (you can either provision a new one or upgrade an existing one). Universal Profiling is currently only available on Elastic Cloud.

> - The workloads (to be profiled) must be running on Linux machines, with x86_64 CPUs, with kernel version >= 4.15 

Reviewer: ```suggestion
- The workloads to be profiled must be running on Linux machines with x86_64 CPUs and kernel version >= 4.15.
```

### Configure data ingestion

To configure data ingestion, click on any menu entry under **Universal Profiling** in the navigation menu.

If this is the first time enabling Universal Profiler on your Cloud deployment,
you will see a popup like the following.

![image setup popup](./img/setup.png)

> Click the button **Setup Universal Profiling** to complete this action. 

Author: Mhmhm I am pretty sure the button contains `Setup` so should we change it also in the UI?

> Click the button **Setup Universal Profiling** to complete this action. 

Reviewer 2: Yep, sorry, that's what I meant to say.

> Click the button **Setup Universal Profiling** to complete this action. 

Author: Updated via https://github.com/elastic/kibana/pull/151828

For the setup to be successful, you need to be logged into Kibana as a user with elevated privileges,

> typically the `elastic` user.

Author: It's one of the pre-defined roles for Elasticsearch in Cloud, can be seen in the Security > Roles section

If you already enabled Universal Profiling on the Cloud deployment and are going to upgrade from a previous version,
see the [upgrade guide](./upgrade.md).
**Warning**: At this stage, upgrading **requires you to wipe all existing profiling data**.

> We recommend upgrading nonetheless, as we have added several improvements and new features in the latest version.

Author: Removing data _could_ be autoamted, but we were planning to document only the manual removal process.
See 09e4087

> We recommend upgrading nonetheless, as we have added several improvements and new features in the latest version.

Reviewer 2: Sounds good. I updated the comment accordingly.

## Install the host-agent

The host-agent profiles your fleet, and you need to install and configure it on every machine that you want to profile.
It needs elevated privileges in order to run (currently: `root` / `CAP_SYS_ADMIN`). 

> To install a host-agent deployment and start gathering data, follow the instructions provided in the page.
> You can always go to the instructions by clicking on the **Add data** button in the top-right corner of the page.

Reviewer 2: ```suggestion
After clicking **Set up Universal Profiling** in the previous step, you'll see the instructions for installing the host-agent.
You can also find these instructions by clicking on the **Add data** button in the top-right corner of the page.
```

Author: Leaving this one pending as we need to clear the UI text first.

### Configure data ingestion

To configure data ingestion, click on any menu entry under **Universal Profiling** in the navigation menu.

If this is the first time enabling Universal Profiler on your Cloud deployment,
you will see a popup like the following.

![image setup popup](./img/setup.png)

Click the button **Setup Universal Profiling** to complete this action. 

> Be aware that for the setup to be successful, you need to be logged into Kibana with a user with elevated privileges,

Reviewer: ```suggestion
For the setup to be successful, you need to be logged into Kibana as a user with elevated privileges,
```

typically the `elastic` user.

> If you had already enabled Universal Profiling on the Cloud deployment and you are upgrading from a previous version,

Reviewer: ```suggestion
If you already enabled Universal Profiling on the Cloud deployment and are going to upgrade from a previous version,
```

see the [upgrade guide](./upgrade.md).

> Please note that, at this stage, upgrading **requires you to wipe all existing profiling data** and start from scratch.

Reviewer: ```suggestion
**Warning**: At this stage, upgrading **requires you to wipe all existing profiling data**.
```

We recommend upgrading nonetheless, as we have added several improvements and new features in the latest version.

## Install the host-agent

The host-agent profiles your fleet, and you need to install and configure it on every machine that you want to profile.
It needs elevated privileges in order to run (currently: `root` / `CAP_SYS_ADMIN`). 

To install a host-agent deployment and start gathering data, follow the instructions provided in the page.
You can always go to the instructions by clicking on the **Add data** button in the top-right corner of the page.

Here are two examples of the provided instructions, for Kubernetes and Binary deployments.

![image add data k8s](./img/add-data-kubernetes.png) ![image add data binary](./img/add-data-binary.png)

**Host-agent configuration notes**

> - The instructions use a `stable` version. This is good for testing environments. For production environments,

Reviewer: Not sure what this means =)

Author: Do you mean that it's missing a piece of sentence or that it is not clear what users should do?

Reviewer: The test says something about a "stable" HA release. How does a customer know what that means ? It implies that there are "unstable" versions available.
And then there are "immutable" versions of HA (for production).

For me this all doesn't make much sense. But how should a beta customer even feel ?

Author: The points below explain how to get immutable versions

Reviewer 2: Maybe we don't need to say stable? 

```suggestion
- The instructions in Kibana work well for testing environments. For production environments,
```

[Elastic container library repository](https://container-library.elastic.co/r/observability/profiling-agent).

- For Kubernetes deployments, the Helm chart version is already used to configure the same container image, unless

> overwritten with the `version` parameter in te Helm values file.

Reviewer: ```suggestion
overwritten with the `version` parameter in the Helm values file.
```

Each tag must match `^[a-zA-Z0-9-:._]+$` regex and use `;` as a separator. Invalid tags are dropped
and warnings issued on startup. 

In Kibana, the KQL field `tags` can be used for filtering. For example, when running host-agent with the following:

```
sudo pf-host-agent/pf-host-agent -project-id=1 -tags='cloud_region:us-central1;env:staging'
```

> in Kibana, you can filter profiling data from the host-agent with the following tag:

Reviewer 2: ```suggestion
You can then filter profiling data from the host-agent in Kibana with the following tag:
```

Each tag must match `^[a-zA-Z0-9-:._]+$` regex and use `;` as a separator. Invalid tags are dropped
and warnings issued on startup. 

> In Kibana, the KQL field `tags` can be used for filtering. For example, when running host-agent with the following:

Reviewer 2: ```suggestion
In Kibana, you can use the KQL field `tags` for filtering. For example, you can run the host-agent with the following:
```
Do we want to say what the result is that you'll get when you run the host-agent with the following?

Author: Probably it's not evident: the result is described just below, it's that in Kibana you can add filters by tags.

https://github.com/elastic/universal-profiling-documentation/pull/29/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R127-R132

On the agent side, there's no difference in running with or without tags

Reviewer 2: Ok, I understand. I updated the suggestion.

```
sudo pf-host-agent/pf-host-agent -project-id=1 -tags='cloud_region:us-central1;env:staging'
```

> in Kibana, you can filter profiling data from the host-agent with:

Reviewer 2: ```suggestion
in Kibana, you can filter profiling data from the host-agent with the following tag:
```
